recipe/meta.yaml

```diff
@@ -22,11 +22,11 @@ requirements:
   run:
     # I don't think any of these constraints are real but here I am
     - python >=3.8
-    - jax
-    - optax =0.1.4
-    - pandas >=1.1,<2a0
-    - dill
-    - tqdm
+    - jax >=0.4.2,<0.5
+    - optax >=0.1.4,<0.2
+    - pandas >=1.1
+    - dill >=0.3.6,<0.4
+    - tqdm >=4.64.1,<5
```

Comment:
Really? This code fails with tqdm greater than 5? I find that hard to believe.

Reply:
Ditto for the other constraints.

Reply:
I don't know myself if the code will or won't work with other versions, these are just the constraints as specified by the author (who is in a better position to judge).

If the constraints are overly tight they should probably be fixed upstream by putting in a PR there and letting the upstream CI check if it breaks or not.

Reply:
I'm pretty sure all the upper bounds are greater than the latest published versions so aren't (yet) affecting anything.

Version 5, when it is published, will be a major version bump which according to semver *will* have backwards incompatible changes. Whether or not those changes will affect `pzflow` is unknowable, but it's certainly much more likely to do so than than a minor version bump.

Some people are against defensive upper bound pins but I personally am in favour as they prevent broken packages from being published, and the cost, in the `conda-forge` ecosystem at least, is simply requiring a PR to bump the pin when the new version has been published (and tested to not break the package).